### ISSUE TITLE
Citation counts based on Current + Jena 200 lists

### ISSUE THREAD

**Issue by nitsnatsnok** [lingdb/CoBL#371]
Paul requested this in [#366](https://github.com/lingdb/CoBL/issues/366#issuecomment-263425510):
> Which brings up a question: are the citation counts based on the `Jena 200` meaning list, and the `Current` language list? Or on all meanings in all languages in the database -- many of them will never be used again, so we are wasting our time on them. Please confirm whether the counts are indeed based on the `Jena 200` meaning list, and the `Current` language list. If not, they should be.

Please change the counts so that they display only these meanings and languages.

**Comment by runjak** [comment]
@khoidt for this it would make the most sense to use the LanguageList and MeaningList as provided by [defaultModels.py](https://github.com/lingdb/CoBL/blob/master/ielex/lexicon/defaultModels.py).

**Comment by khoidt** [comment]
Do I understand correctly that the "counts" are the `Source`-related models (`CognateJudgement`, `CognateClass`, and `Lexeme`), i.e.:
![image](https://cloud.githubusercontent.com/assets/7487683/20886484/781e5386-baf6-11e6-9185-2fb5f237b327.png)
and they should be counted/included in `Source`-related views only if their related `Language` is in `LanguageList.DEFAULT`?

**Comment by nitsnatsnok** [comment]
Do you speak English? :wink:
As far as I understand this you're right (so yes, your screenshot actually shows what we called “counts”), but additionally their meaning has to be in the Jena 200 list, therefore include them only if:
- their related `Language` is in `LanguageList.DEFAULT` **and**
- their related meaning is in the Jena 200 list (for which I unfortunately have no clue how you call it)

I opened this as an issue now because I do a lot of work based on these counts (like in #365), and understandably Paul doesn't want me to waste time on languages/meanings we don't use.

**Comment by khoidt** [comment]
Pythonic English (CoBL dialect), perhaps, to avoid misunderstandings when translating this issue to code 😉. @runjak, can you tell me where do I find the Jena 200 list?

**Comment by runjak** [comment]
The `.DEFAULT` fields are a good start here, but it would be better to use the functions in from `defaultModels.py` which give the currently selected lists.
These lists are the defaults unless the user changes them and are stored in the users session.
Using the provided functions would make sure that counts would still be as expected should the user change the lists.

Also yes, @khoidt you are correct - we usually want to restrict lexemes, meanings and languages to the currently selected default lists.

TL;DR: Please use [getDefaultLanguage(…)](https://github.com/lingdb/CoBL/blob/master/ielex/lexicon/defaultModels.py#L13) and [getDefaultMeaning(…)](https://github.com/lingdb/CoBL/blob/master/ielex/lexicon/defaultModels.py#L61).

**Comment by nitsnatsnok** [comment]
@khoidt I don't want to rush you, but are you able to get this working in the next days? #365 depends on it…

**Comment by khoidt** [comment]
I'm on it right now. Hopefully, will be included in today's update.

**Comment by khoidt** [comment]
With the last update, the counts are now filtered as requested here. Should the related views also be filtered? If not, the issue can be closed after deployment and testing of the last commit (d4db9f2).

**Comment by runjak** [comment]
:rocket: deployed.

**Comment by nitsnatsnok** [comment]
> Should the related views also be filtered?

If by this you mean the tables on cobl.info/sources/_ID_/{cognacy,cogset,lexeme}/, then yes please.

@khoidt However, the tables doesn't seem to work at all currently, cf.:
![2016-12-20-101749_1366x768_scrot](https://cloud.githubusercontent.com/assets/2664162/21344833/11d572ea-c69e-11e6-901f-45354138fd66.png)
These are the counts of source 156, but the linked pages cobl.info/sources/156/{cognacy,cogset,lexeme}/ don't display anything.
Should I file a bug for this?

**Comment by khoidt** [comment]
Hmm. Locally it works as it should, e.g. cognacy for source 156:
![image](https://cloud.githubusercontent.com/assets/7487683/21361256/8d411264-c6e3-11e6-9da9-8bb8dece234b.png)
@runjak, might it have something to do with the last update? (I still haven't pulled the latest version).

**Comment by runjak** [comment]
I'll check on this in a minute.

**Comment by runjak** [comment]
For me the mentioned page looks like this:
![screenshot_2016-12-21_14-41-27](https://cloud.githubusercontent.com/assets/5417642/21391339/9782484e-c78b-11e6-930f-47eadf21c0c3.png)
I'm currently debugging where this could come from.

I've found a problem with the JavaScript code, which in current django didn't find the jquery instance (especially not when the debug toolbar doesn't bring it's own version separate from ours). I solved this by refactoring the JavaScript code into a dedicated script in c36e30fb1c31a347a486a1c571dbf5bfd374c943. With this code these buttons for citations work on my machine:
![screenshot_2016-12-21_14-43-27](https://cloud.githubusercontent.com/assets/5417642/21391403/e52cf8a0-c78b-11e6-9767-f24e90950c18.png)

I'll update here once I've figured out what goes wrong with the tables on my setup.

**Comment by runjak** [comment]
All three of the problematic pages appear to be rendered with help of the [source_related_inline.html](https://github.com/lingdb/CoBL/blob/fb9e7928c642103468bc164473b3c92abe305ff5/ielex/templates/source_related_inline.html) template.
I'm not sure why no data is being displayed because there are also no error messages in the logs for this and the code appears to be running without problems.

@khoidt I don't think [source_related_inline.html:34-51](https://github.com/lingdb/CoBL/blob/fb9e7928c642103468bc164473b3c92abe305ff5/ielex/templates/source_related_inline.html#L34-L51) actually produces valid html as there can easily be multiple `</thead><tbody>` without closing the `<tbody>` or reopening the `<thead>` in between. Also the `<th>` inside the `<thead>` should still belong in a `<tr>` I think.

For further analysis I would suggest working from the most recent `devel` (merging that into branch `306`) to be able to reproduce the situation.

**Comment by runjak** [comment]
Deployed.
